TrackingTools/TransientTrack/src/TransientTrackBuilder.cc

```diff
@@ -30,10 +34,24 @@ TransientTrack TransientTrackBuilder::build (const GsfTrack & t) const {
 }
 
 TransientTrack TransientTrackBuilder::build (const CandidatePtr * t) const {
+  reco::PFCandidatePtr tryPF(*t);
+  edm::Ptr<pat::PackedCandidate> tryPacked(*t);
+  if( tryPF.get() != nullptr && tryPF->isTimeValid() ) {
+    return TransientTrack(*t, tryPF->time(), tryPF->timeError(), theField, theTrackingGeometry);
```

Comment:
what is the cost of this casting?
Igprof should capture this well enough

Reply:
given that reco::Candidate already has dzError and dxyError in the interface, it's more productive to add time (or vt) and tError to the reco::Candidate interface than to impose this casting.

Reply:
There is little to no penalty for this casting. build(CandidatePtr *t) is not used in any high bandwidth applications (like track reconstruction for instance).

Reply:
However, I see your point. *But* I would suggest we include a change to something as deep in CMSSW as reco::Candidate for a different PR.